src/Altinn.App.PlatformServices/Implementation/ApplicationLanguage.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.App.PlatformServices.Helpers;
+using Altinn.App.PlatformServices.Interface;
+using Altinn.App.Services.Configuration;
+using Altinn.App.Services.Implementation;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.PlatformServices.Implementation
+{
+    /// <summary>
+    /// An implementation used to retrieve the supported application languages.
+    /// </summary>
+    public class ApplicationLanguage : IApplicationLanguage
+    {
+        private readonly AppSettings _settings;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationLanguage"/> class.
+        /// </summary>
+        /// <param name="settings">The app repository settings.</param>
+        /// <param name="logger">A logger from the built in logger factory.</param>
+        public ApplicationLanguage(
+            IOptions<AppSettings> settings,
+            ILogger<AppResourcesSI> logger)
+        {
+            _settings = settings.Value;
+            _logger = logger;
+        }
+
+        /// <inheritdoc />
+        public async Task<List<Platform.Storage.Interface.Models.ApplicationLanguage>> GetApplicationLanguages()
+        {
+            string pathTextsResourceFolder = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.TextFolder;
```

Comment:
Instead of concatenating strings, leaving it open for directory separator character issues (/ or \ or nothing), use [Path.Join](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.join?view=net-6.0) instead.